### PR TITLE
feat: add docker build and CI

### LINKED ISSUE #123: Sentinel Deploy: Publish Custom Telegraf Docker Image
In order to capture custom Telegraf deployable for k8s, we need to prepare https://github.com/filecoin-shipyard/telegraf-lotus/ as a built docker image.

- [x] Create a repository in Dockerhub
- [x] Create or modify Dockerfile to build image
- [x] Build and Publish docker image manually
- [x] Stretch Goal: build and publish docker image in CI

### PR DESCRIPTION
**Pull request by olizilla**
Adds a docker image for `sentinel-drone`, and brings back CI.

- Adds a Dockerfile at the root of the project. It's telegrafs own dockerfile, customised with build deps for `filecoin-ffi`.
- Tweak the Makefile to make the docker build work, and to limit the scope of testing to just our custom plugins, so we can get back to testing in CI.
- Adds a circleci config. A stripped back version of telegrafs original circle config, with an extra docker-build target. Gets us running `gofmt`, `go vet`, and `go test` on PRs.

💚 https://app.circleci.com/pipelines/github/filecoin-project/sentinel-drone/12/workflows/f4cc3834-5f56-4c6f-9476-4cf5d779dd4a

Screenshot 2020-09-14 at 16 04 28

fixes https://github.com/filecoin-project/sentinel/issues/123

### TODO

- [x] get circle ci running again
- [x] customise docker build target
- [x] get docker build to work
- [x] publish docker image to hub
- [ ] ~~publish docker image to hub from ci~~

dockerhub has been configured to watch the repo and autobuild the latest image when master changes.

License: MIT
Signed-off-by: Oli Evans <oli@tableflip.io>

### CODE REVIEW

**Comment by olizilla** [comment]
Ok, this is worth a review... the failure in ci is coming from _drum roll_ a test failure in one of **our** tests

```
--- FAIL: TestTagMetricWithPeerID (0.00s)
    assert.go:15: /go/src/github.com/influxdata/telegraf/plugins/processors/lotus_peer_id/peerid_test.go:40
    assert.go:24: ! ["lotus_peer_id"]: "12D3KooWRHNnsJmKEsJwqxesunwuQXS4V7n74PoK6Ca5NJ87bUcu" != (missing)
    assert.go:24: ! ["peer_id"]: (missing) != "12D3KooWRHNnsJmKEsJwqxesunwuQXS4V7n74PoK6Ca5NJ87bUcu"
FAIL
```

**Comment by olizilla** [comment]
Now we just need #28 to fix our test error and CI should turn green!

**Comment by olizilla** [comment]
For auto-publish to dockerhub, I can enable dockerhub auto build. One problem we had with that in the past was not knowing when the dockerhub build failed, but this PR adds in a docker-build job so we'll see on the PR when that fails. Alternatively we can have circleci push the build to dockerhub, but that requires putting dockerhub creds in circle. Having dockerhub watch the repo is mildly more "secure" depending on your threat model.

**Comment by olizilla** [comment]
I've set up dockerhub to watch the repo. It'll autobuild the `latest` tag when `master` changes. We can set docker version tags by tagging the repo with anything that matches anything semver like that starts with a `v` (matching `/^v([0-9.]+)$/`); a git tag `v1` would create a docker tag `v1`.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          name: Build Docker image
+          command: |
+            docker build -t $IMAGE_NAME:$WIP_IMAGE_TAG .
+      - run:
+          name: Archive Docker image
+          command: docker save -o docker-image.tar $IMAGE_NAME
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./docker-image.tar
```
Comment (placer14) [inline .circleci/config.yml]: Just out of curiousity, why is this valuable? I don't see it consumed by anything obvious.

**Review by placer14 — APPROVED** [review]
This is excellent stuff! 🙇 Will leave available for a look from @ognots. Happy to merge when ready.

File: .circleci/config.yml (hunk 1 above)
Comment (olizilla) [reply .circleci/config.yml]: I was hedging my bets on how to automate publishing the image. This line is there so another CI job could push it to docker hub for a blessed branch, as we do in go-ipfs, but now we've got dockerhub watching the repo it's not needed. 

For context, we had issues with dockerhub builds failing for go-ipfs, so we moved the entire process into our CI. In this case, as we have a docker-build job that will tell us when the image can't be built, I'm taking a leap of faith, and trusting dockerhub again.

I'll remove this vestigial nub.

**Comment by olizilla** [comment]
I am merging this. Can always fix any issues in a new PR.

**Pull request merged**